packages/shared_preferences/shared_preferences/example/test/example_app_test.dart

```diff
@@ -23,3 +24,100 @@ void main() {
     });
   });
 }
+
+// Note: this code is duplicated in
+// shared_preferences/test/shared_preferences_async_test.dart. Since we cannot
+// import the relative path ../../test/shared_preferences_async_test.dart on the
+// web platform, we had to copy it here for use in this test library.
```

Comment:
Is there a way to put this code somewhere that can be imported in both locations that isn't exported to users?

Reply:
Short of creating a shared package for testing utilities, I'm not sure how else to share this code.

I'm not entirely sure why the relative path can be imported successfully from a test for native platforms but not for the web platform. CC @jakemac53 do you know if this behavior is expected from the test package for this case? 

Another option is to add `TestOn('vm')` here so that this test does not run for the web platform.

Reply:
This is expected yes, the way apps are hosted on the web we have no way to access URIs outside of the package.

The "root" of the server is either the test directory or the package directory (can't remember which), so you can only access things from the current package and then package: URIs (via synthetic /packages/ paths).

Reply:
These sorts of relative paths also really shouldn't be used anyways, they can easily be broken (that file could have dependencies you don't have for instance).

The solution is a shared testing package (no need to publish it).

Reply:
@tarrinneal are you okay landing this with a todo to create a shared testing package or do we need to do that as part of this PR?